pkg/util/http.go

```diff
@@ -329,7 +329,7 @@ func ErrorTypeFromHTTPStatus(status int) string {
 }
 
 func IsError(status int) bool {
-	return status/200 != 0
+	return status/100 != 2
```

Comment:
I think that this code is still needlessly obtuse. I'd rewrite this as 
```suggestion
	return status < 200 || status >= 300
```

It's probably faster and is definitely more understandable. Also, technically `1xx` status codes are not errors, so a more accurate name of this function would be `IsNotSuccess`

Reply:
Then, why not call it `IsSuccess()` and use `if !IsSuccess() { ... }`

```
func IsSuccess(status int) bool {
  return status / 100 == 2
}

Reply:
I can rename the method if required, although `if IsError { retry... }` sounded fairly readable too